### PR TITLE
Cli: Decode known program custom errors returned from simulation

### LINKED ISSUE #17637: CLI doesn't decode transaction simulation errors
#### Problem
CLI actions may return cryptic error codes, even though the CLI knows how to decode the errors for all the programs it supports. eg:
```
Error: RPC response error -32002: Transaction simulation failed: Error processing Instruction 0: custom program error: 0x7 [2 log messages]
```

This is due to the addition of transaction simulation to the processing workflow on rpc nodes. The CLI will decode actual instruction errors returned by tx confirmation, but hasn't been updated to handle simulation errors.

#### Proposed Solution
Parse simulation errors to find and decode InstructionErrors

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
CLI actions may return cryptic error codes, even though the CLI knows how to decode the errors for all the programs it supports.

#### Summary of Changes
Add handling for RpcSimulateTransactionResult > InstructionError > Custom

Fixes #17637

### CODE REVIEW

File: cli/src/cli.rs [hunk 1]
```diff
@@ -1956,6 +1957,28 @@ where
...
+                            err:
+                                Some(TransactionError::InstructionError(
+                                    _,
+                                    InstructionError::Custom(code),
+                                )),
+                            ..
+                        },
+                    ),
+                ..
+            }) = err.kind()
```
Comment (t-nelson) [inline cli/src/cli.rs:1974]: That's a mouthful! :joy:

**Review by t-nelson — APPROVED** [review]
lgtm!

File: cli/src/cli.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli/src/cli.rs:1974]: Heh, I know! 😳

**Comment by CriesofCarrots** [comment]
CI error is unrelated (#17655 )

**Pull request merged**